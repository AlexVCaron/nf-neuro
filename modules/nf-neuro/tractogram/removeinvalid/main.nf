process TRACTOGRAM_REMOVEINVALID {
    tag "$meta.id"
    label 'process_single'

    container "scilus/scilpy:2.2.0_cpu"

    input:
        tuple val(meta), path(tractogram)

    output:
        tuple val(meta), path("*.{trk,tck}"), emit: tractograms
        path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def suffix = task.ext.suffix ? "_${task.ext.suffix}" : ""

    def cut_invalid = task.ext.cut_invalid ? "--cut_invalid" : ""
    def remove_single_point = task.ext.remove_single_point ? "--remove_single_point" : ""
    def remove_overlapping_points = task.ext.remove_overlapping_points ? "--remove_overlapping_points" : ""
    def threshold = task.ext.threshold ? "--threshold " + task.ext.threshold : ""
    def no_empty = task.ext.no_empty ? "--no_empty" : ""

    """
    for tractogram in ${tractogram};
        do \
        ext=\${tractogram#*.}
        bname=\$(basename \${tractogram} .\${ext})

        scil_tractogram_remove_invalid \$tractogram ${prefix}__\${bname}${suffix}.\${ext}\
                        $cut_invalid\
                        $remove_single_point\
                        $remove_overlapping_points\
                        $threshold\
                        $no_empty -f
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def suffix = task.ext.suffix ? "_${task.ext.suffix}" : ""

    """
    scil_tractogram_remove_invalid -h

    for tractogram in ${tractogram};
        do \
        ext=\${tractogram#*.}
        bname=\$(basename \${tractogram} .\${ext})

        touch ${prefix}__\${bname}${suffix}.\${ext}
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
