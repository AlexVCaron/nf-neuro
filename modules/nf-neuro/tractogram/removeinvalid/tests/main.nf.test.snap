{
    "tractogram - remove invalid - stub-run": {
        "content": [
            [
                "versions.yml:md5,e02f390445832d14996dfdf011cc3fd7"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-23T00:32:33.981463851"
    },
    "tractogram - remove invalid": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__bundle_0.trk:md5,6fb44e5444b09eaf6e612e1bde04b0ed",
                            "test__bundle_1.trk:md5,04fbc7404a45e197d203d05e1bde351a",
                            "test__bundle_2.trk:md5,91cf2d5b1e8ca0e219012dbcaf4a8edb",
                            "test__bundle_3.trk:md5,a2b56e397fc062691ed4c99f8fe67228",
                            "test__bundle_4.trk:md5,3fe0cd679a23e53079bcfdfb3a777e89",
                            "test__bundle_5.trk:md5,9a7d5da64e557eb6622a051593886e4a",
                            "test__bundle_6.trk:md5,29434c0aa3bc72c721e8879edd4f0dca"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,e02f390445832d14996dfdf011cc3fd7"
                ],
                "tractograms": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__bundle_0.trk:md5,6fb44e5444b09eaf6e612e1bde04b0ed",
                            "test__bundle_1.trk:md5,04fbc7404a45e197d203d05e1bde351a",
                            "test__bundle_2.trk:md5,91cf2d5b1e8ca0e219012dbcaf4a8edb",
                            "test__bundle_3.trk:md5,a2b56e397fc062691ed4c99f8fe67228",
                            "test__bundle_4.trk:md5,3fe0cd679a23e53079bcfdfb3a777e89",
                            "test__bundle_5.trk:md5,9a7d5da64e557eb6622a051593886e4a",
                            "test__bundle_6.trk:md5,29434c0aa3bc72c721e8879edd4f0dca"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e02f390445832d14996dfdf011cc3fd7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-23T00:32:21.665134954"
    }
}